src/main/java/com/synopsys/integration/jira/common/cloud/rest/service/JiraAppService.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * int-jira-common
+ *
+ * Copyright (c) 2019 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.jira.common.cloud.rest.service;
+
+import org.apache.commons.codec.Charsets;
+import org.apache.commons.codec.binary.Base64;
+
+import com.google.gson.Gson;
+import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.jira.common.cloud.model.request.AppUploadRequestModel;
+import com.synopsys.integration.jira.common.cloud.rest.JiraCloudHttpClient;
+import com.synopsys.integration.rest.HttpMethod;
+import com.synopsys.integration.rest.body.StringBodyContent;
+import com.synopsys.integration.rest.request.Request;
+import com.synopsys.integration.rest.request.Response;
+
+public class JiraAppService {
+    public static final String API_PATH = "/rest/plugins/1.0/";
+
+    private Gson gson;
+    private JiraCloudHttpClient httpClient;
+    private JiraCloudService jiraCloudService;
+
+    public JiraAppService(final Gson gson, JiraCloudHttpClient httpClient, final JiraCloudService jiraCloudService) {
+        this.gson = gson;
+        this.httpClient = httpClient;
+        this.jiraCloudService = jiraCloudService;
+    }
+
+    public Response installApp(String pluginName, String pluginUri, String username, String accessToken) throws IntegrationException {
+        String uri = jiraCloudService.getBaseUrl();
+        final String pluginToken = retrievePluginToken(uri, username, accessToken);
+        final Request request = createUploadRequest(uri + API_PATH, username, accessToken, pluginToken, pluginName, pluginUri);
+        return httpClient.execute(request);
+    }
+
+    public Response uninstallApp(String appKey, String username, String accessToken) throws IntegrationException {
+        String apiUri = jiraCloudService.getBaseUrl() + API_PATH;
+        final String pluginToken = retrievePluginToken(apiUri, username, accessToken);
+        final Request request = createDeleteRequest(apiUri + appKey + "-key", username, accessToken, pluginToken);
```

Comment:
Is there any guarantee that the apiUri is going to end in a '/'? The string concatenation to create the URLs seems questionable